src/zope/app/container/browser/tests/__init__.py

```diff
@@ -1,2 +1,60 @@
 #
-# This file is necessary to make this directory a package.
+
+import unittest
+
+from webtest import TestApp
+
+
+
+class BrowserTestCase(unittest.TestCase):
+
+    # def setSite(self, site):
+    #     """Set the site which will be used to look up local components"""
+    #     setSite(site)
+
+
+    # def getSite(self):
+    #     """Returns the site which is used to look up local components"""
+    #     return getSite()
+
+    def getRootFolder(self):
+        return self.layer.getRootFolder()
+
+    def setUp(self):
+        super(BrowserTestCase, self).setUp()
+        self._testapp = TestApp(self.layer.make_wsgi_app())
+        # Typically this would be done by zope.app.principalannotation's
+        # bootstrap.zcml but we don't have a dep on that.
+        from zope.principalannotation.utility import PrincipalAnnotationUtility
+        from zope.principalannotation.interfaces import IPrincipalAnnotationUtility
+        from zope import component
+        component.getGlobalSiteManager().registerUtility(PrincipalAnnotationUtility(),
+                                                         IPrincipalAnnotationUtility)
+
+    def publish(self, path, basic=None, form=None, env={},
```

Comment:
Sorry. This whole method was directly copied from [`z.a.testing.functional. BrowserTestCase'](https://github.com/zopefoundation/zope.app.testing/blob/master/src/zope/app/testing/functional.py#L474) and then just modified to make it work. I agree I can do better.